pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,53 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	var instanceID string
+	// As the CSR has been approved get the Kubernetes node object associated
+	// the Windows VM created
+	if nc.Windows.GetCredentials() != nil && len(nc.Windows.GetCredentials().GetInstanceId()) > 0 {
+		instanceID = nc.Windows.GetCredentials().GetInstanceId()
+	}
+
+	// Get the node object associated with node
+	node, err := nc.getNodeObjectAssociated(instanceID)
+	if err != nil {
+		return errors.Wrap(err, "node object associated with the node")
+	}
+
+	// Apply worker labels
+	if err := nc.applyWorkerLabel(node); err != nil {
+		return errors.Wrap(err, "failed applying worker label")
+	}
+	return nil
+}
+
+func (nc *nodeConfig) getNodeObjectAssociated(instanceID string) (*v1.Node, error) {
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node.openshift.io/os_id=Windows"})
```

Comment:
Why are you have `node.openshift.io/os_id` as string? Are you having a cyclic dependency causing this problem?

Reply:
yes

Reply:
Then we need to find a better place for the constant. Adding it as a string is not acceptable.

@ravisantoshgudimetla we might have to move away from the sub-packages to make our lives easier.